bpo-25438: document what codec PyMemberDef T_STRING decodes the char * as

Windsooon opened the pull request:
Source of T_STRING: https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Python/structmember.c#L51

Source of PyUnicode_FromString
https://github.com/python/cpython/blob/master/Include/unicodeobject.h#L702

<!-- issue-number: [bpo-25438](https://bugs.python.org/issue25438) -->
https://bugs.python.org/issue25438
<!-- /issue-number -->

Windsooon committed:
document what codec PyMemberDef T_STRING decodes the char * as

lysnikolaou commented on Doc/c-api/structures.rst:
```diff
@@ -292,7 +292,7 @@ definition with the same method name.
    :attr:`flags` can be ``0`` for write and read access or :c:macro:`READONLY` for
    read-only access.  Using :c:macro:`T_STRING` for :attr:`type` implies
-   :c:macro:`READONLY`.  Only :c:macro:`T_OBJECT` and :c:macro:`T_OBJECT_EX`
+   :c:macro:`READONLY` and T_STRING will always interpreted as UTF-8 encoded string.  Only :c:macro:`T_OBJECT` and :c:macro:`T_OBJECT_EX`
```
Could you please include the `:c:macro:` directive as is done in the preceding lines? It would be also good to wrap the line around 80 columns, in order to be consistent with the rest of the file.

gpshead committed:
tiny grammar fix (pluralize is -> are)

gpshead committed:
addressing review comment

adds missing :c:macro: annotation, splits the run on sentence, and limits line length.

gpshead committed:
undo change in this file

unrelated to the issue, and "multiple assignment" not plural is considered to be a single thing in the contexts I've heard it used.
